pkg/apis/stringer/v1alpha1/stringer.go

```diff
@@ -35,7 +35,7 @@ func Yaml(ctx string, obj interface{}) string {
 	}
 	b, err := yaml.Marshal(obj)
 	if err != nil {
-		return fmt.Sprintf("\n%s {%+v}", ctx, obj)
+		return fmt.Sprintf("\n%s {--failed to get details--}", ctx)
```

Comment:
If the obj is already of type string then we can still print the object without introducing cyclic dependency.

Can below code work better?
Please introduce additional UT as well.

```go
if obj == nil {
  return fmt.Sprintf("\n%s {nil}", ctx)
}

str, ok := obj.(string)
if ok {
  return fmt.Sprintf("\n%s {%s}", ctx, str)
}

b, err := yaml.Marshal(obj)
if err != nil {
  return fmt.Sprintf("\n%s {--failed to get details--}", ctx)
}

return fmt.Sprintf("\n%s {%s}", ctx, string(b))
```